### ISSUE TITLE
FEAT: Add a comment to justify a choice that ignores a warning

### ISSUE THREAD

**Issue by sylvaincom** [probabl-ai/skore#1039]
### Is your feature request related to a problem? Please describe.

As suggested by @ogrisel 

For now (skore v0.5), in `skore.train_test_split`, we just raise a warning in the notebook when we believe that there is a methodological mistake that is made by the user.

### Describe the solution you'd like

It would be nice to automatically ask the users to add a comment in the notebook explaining why they set `shuffle=False` for `train_test_split` for example (which is not the default recommended setting), so that another user can understand this choice and even the user itself in a few months might forget. The idea is to document any "counter-intuitive" / "non obvious" choice.

This would be true for all warnings, not just for `skore.train_test_split`.

**Comment by MarieS-WiMLDS** [comment]
Isn't it a duplicate of this one: https://github.com/probabl-ai/skore/issues/497?  
This epic was closed because we know that the UX for warnings are not great today, we should change it, therefore it seems weird to keep working on improving something we know we will change.
